src/utils.rs

```diff
@@ -50,3 +58,35 @@ pub fn parse_gj_messages_response(meow: String) -> Result<Vec<GDMessage>, Generi
         })
         .collect()
 }
+
+pub fn valid_url_check(url: &str) -> bool {
+    let reg = regex::Regex::new(r#"^(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$"#).unwrap();
```

Comment:
this regex should be [case insensitive](https://docs.rs/regex/1.3.1/regex/#grouping-and-flags), see below comment:
```suggestion
    let reg = regex::Regex::new(r#"(?i)^(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$"#).unwrap();
```